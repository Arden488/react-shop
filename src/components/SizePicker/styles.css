.size-picker {
  display: grid;
  grid-template-columns: repeat(5, auto);
  grid-column-gap: 20px;
}

.size-picker__item {
  border: 1px solid #666;
  border-radius: 4px;
  min-height: 60px;
  text-align: center;
}

.size-picker__item_selected {
  border-color: #ff9900;
}

.size-picker__label {
  font-size: 16px;
  display: block;
  margin: auto;
}

.size-picker__input {
  display: none;
}
