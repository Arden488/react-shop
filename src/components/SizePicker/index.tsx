import * as React from 'react';

import './styles.css';

const SizePicker = () => (
  <div className="size-picker">
    <button type="button" className="size-picker__item size-picker__item_selected">
      <span className="size-picker__label">16 Gb</span>
      <input className="size-picker__input" type="checkbox" name="" />
    </button>
    <button type="button" className="size-picker__item">
      <span className="size-picker__label">32 Gb</span>
      <input className="size-picker__input" type="checkbox" name="" />
    </button>
    <button type="button" className="size-picker__item">
      <span className="size-picker__label">128 Gb</span>
      <input className="size-picker__input" type="checkbox" name="" />
    </button>
    <button type="button" className="size-picker__item">
      <span className="size-picker__label">256 Gb</span>
      <input className="size-picker__input" type="checkbox" name="" />
    </button>
    <button type="button" className="size-picker__item">
      <span className="size-picker__label">512 Gb</span>
      <input className="size-picker__input" type="checkbox" name="" />
    </button>
  </div>
);

export default SizePicker;
