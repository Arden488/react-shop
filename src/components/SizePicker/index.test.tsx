import { shallow } from 'enzyme';
import * as React from 'react';

import SizePicker from './index';

let component: React.ReactNode = null;

beforeEach(() => {
  component = shallow(<SizePicker />);
});

it('renders correctly', () => {
  expect(component).toMatchSnapshot();
});
