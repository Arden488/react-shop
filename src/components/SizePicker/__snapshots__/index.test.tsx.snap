// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="size-picker"
>
  <button
    className="size-picker__item size-picker__item_selected"
    type="button"
  >
    <span
      className="size-picker__label"
    >
      16 Gb
    </span>
    <input
      className="size-picker__input"
      name=""
      type="checkbox"
    />
  </button>
  <button
    className="size-picker__item"
    type="button"
  >
    <span
      className="size-picker__label"
    >
      32 Gb
    </span>
    <input
      className="size-picker__input"
      name=""
      type="checkbox"
    />
  </button>
  <button
    className="size-picker__item"
    type="button"
  >
    <span
      className="size-picker__label"
    >
      128 Gb
    </span>
    <input
      className="size-picker__input"
      name=""
      type="checkbox"
    />
  </button>
  <button
    className="size-picker__item"
    type="button"
  >
    <span
      className="size-picker__label"
    >
      256 Gb
    </span>
    <input
      className="size-picker__input"
      name=""
      type="checkbox"
    />
  </button>
  <button
    className="size-picker__item"
    type="button"
  >
    <span
      className="size-picker__label"
    >
      512 Gb
    </span>
    <input
      className="size-picker__input"
      name=""
      type="checkbox"
    />
  </button>
</div>
`;
